design/authn_azure/authn_azure_solution_design.md

```diff
@@ -0,0 +1,474 @@
+VMs with Conjur by creating a Conjur host identity for their Azure resources. The Azure resource would then receive an API key as part of the 
+creation of a Conjur identity which they must use to connect to targets. This is not optimal nor convenient because we are not allowing the 
+resources to authenticate based on their Azure identity and demand that the resource hold an extra secret, the Conjur-given API key.
+
+## Solution
+
+We will add a new authenticator to Conjur, which users can authenticate with from Azure VMs. As mentioned in the feature doc, before any authentication 
+request is sent to Conjur, the admin will load the authenticator policy: 
+
+```yaml
+# policy id needs to match the convention `conjur/authn-azure/<service ID>`
+- !policy
+  id: conjur/authn-azure/prod
+  body:
+  - !webservice
+  
+  - !variable
+    id: provider-uri
+      
+  - !group apps
+  
+  - !permit
+    role: !group apps
+    privilege: [ read, authenticate ]
+    resource: !webservice
+```
+
+and load a value to the `provider-uri` variable (e.g `https://sts.windows.net/TENANT_ID`). 
+
+### Azure Application Identity
+
+To authenticate with Conjur, Azure-specific fields will need to be provided in the host annotations of the Conjur host identity for the Azure resource- subscription-id, resource-group, and user/system-assigned-identity. Specifically, If the Azure resource is assigned a system-assigned-identity, 
+we will also need to validate the `oid` field of the Azure access token. These fields will be used to confirm the identity of the Azure resource that is
+sending the authentication request. This will be done by comparing the Azure access token information with the host annotations that are defined in the policy below.
+
+A Conjur host will be defined as follows with their annotations holding Azure-specific identification properties.
+
+```yaml
+- !policy 
+  id: azure-apps 
+  body: 
+  - !host 
+    id: test-app 
+    annotations: 
+      authn-azure/subscription-id: test-subscription 
+      authn-azure/resource-group: test-group 
+      authn-azure/user-assigned-identity: test-app-pipeline 
+OR 
+      authn-azure/system-assigned-identity: 853b9a84-5bfa-4b22-a3f3-0b9a43d9ad8a
+
+- !grant
+  role: !group conjur/authn-azure/prod/apps
+  member: !host azure-apps/test-app
+```
+
+Note that such a host can have only one of the annotations 
+`authn-azure/user-assigned-identity` & `authn-azure/system-assigned-identity`
+
+#### Multiple-assigned identity for host - NOT FOR NOW
+
+We may allow hosts to be defined with both "authn-azure/user-assigned-identity" and "authn-azure/system-assigned-identity". This way such a host can 
+authenticate a token with a user-assigned id as well as a token with a system-assigned id. As a first step we will not develop - nor test - such a use-case.
+
+### Design
+
+#### Authentication Flow
+
+This flow assumes that the VM already exists in Azure, and that an authn-azure authenticator and a Conjur host were loaded.
```

Comment:
Worth mentioning that while we expect the app to be on the VM we currently won't officially support having conjur or the DAP follower in Azure

Reply:
@sigalsax ^

Reply:
No need to mention the DAP Follower (but it's ok to mention it if you think it's better)